src/Mvc/test/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.Test/ChecksumValidatorTest.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Hosting;
+using Microsoft.AspNetCore.Razor.Language;
+using Xunit;
+using static Microsoft.AspNetCore.Razor.Hosting.TestRazorCompiledItem;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation
+{
+    public class ChecksumValidatorTest
+    {
+        private VirtualRazorProjectFileSystem ProjectFileSystem { get; } = new VirtualRazorProjectFileSystem();
+
+        [Fact]
+        public void IsRecompilationSupported_NoChecksums_ReturnsFalse()
+        {
+            // Arrange
+            var item = new TestRazorCompiledItem(typeof(string), "mvc.1.0.view", "/Views/Home/Index.cstml", new object[] { });
+
+            // Act
+            var result = ChecksumValidator.IsRecompilationSupported(item);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void IsRecompilationSupported_NoPrimaryChecksum_ReturnsFalse()
+        {
+            // Arrange
+            var item = new TestRazorCompiledItem(typeof(string), "mvc.1.0.view", "/Views/Home/Index.cstml", new object[]
+            {
+                new RazorSourceChecksumAttribute("SHA1", GetChecksum("some import"), "/Views/Home/_ViewImports.cstml"),
```

Comment:
This isn't about checksums. It's about primary content. Suggest renaming the test.